### PR TITLE
61 implement a split screen layout design

### PR DESCRIPTION
**Pull request by patates-cipsi418**
Developing a module to implement the new split style featured on the robots page.
Enhancing the jumbotrons to support multiple videos.
Substituting the sections on the robots page that previously contained multiple videos with the new jumbotrons components.

### CODE REVIEW

File: src/components/JumbotronVideoComponent.vue [hunk 1]
```diff
@@ -81,4 +101,14 @@ if (props.img) {
...
+
+.split {
+  position: absolute;
+  top: 0;
+  left: 0;
+  height: 100%;
+  width: 100%;
+}
+
+@import '@clubcapra/styles/split.scss';
```
Comment (Leuchak) [inline src/components/JumbotronVideoComponent.vue:113]: place the `import` statement at the top of the <style> tag

File: src/components/JumbotronVideoComponent.vue [hunk 2]
```diff
@@ -9,7 +9,13 @@ const props = defineProps<{
   /**
    * If set, the main element will have a background video.
    */
-  video?: string;
+  videos?: string[];
+  /**
+   * Set the orientation of the spliting.
+   * For the split module
+   * @defaultValue `'vertical'`
```
Comment (Leuchak) [inline src/components/JumbotronVideoComponent.vue:16]: \@ defaultvalue is not valid tsdoc

File: src/components/JumbotronVideoComponent.vue [hunk 3]
```diff
@@ -9,7 +9,13 @@ const props = defineProps<{
   /**
    * If set, the main element will have a background video.
    */
-  video?: string;
+  videos?: string[];
```
Comment (Leuchak) [inline src/components/JumbotronVideoComponent.vue:12]: Here, you have two features: adding multiple videos and the split-screen layout. This should go into its own feature!

File: src/components/JumbotronVideoComponent.vue [hunk 4]
```diff
@@ -9,7 +9,13 @@ const props = defineProps<{
...
+  /**
+   * Set the orientation of the spliting.
+   * For the split module
+   * @defaultValue `'vertical'`
+   */
+  splitOrientation?: 'horizontal' | 'vertical';
```
Comment (Leuchak) [inline src/components/JumbotronVideoComponent.vue:18]: Since there are only two possible values, and there's a default value, I propose transforming the property into a boolean. For example, `isHorizontal`. If the prop is `true`, the layout should be displayed horizontally; otherwise, it's vertical. This should simplify the code and improve readability.

*It's a proposal, the current solution also works well and can easily be modified to add other modes if needed :P

File: src/components/JumbotronVideoComponent.vue [hunk 5]
```diff
@@ -67,7 +85,9 @@ if (props.img) {
   /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
   background: linear-gradient(to right, #000, #121212);
 }
-
+.jumbotron > * {
+  z-index: 0;
+}
```
Comment (Leuchak) [inline src/components/JumbotronVideoComponent.vue:90]: Looks like a hack. what's it for?

File: src/components/JumbotronVideoComponent.vue [hunk 6]
```diff
@@ -81,4 +101,14 @@ if (props.img) {
...
   margin-right: auto;
 }
+
+.split {
+  position: absolute;
+  top: 0;
+  left: 0;
+  height: 100%;
+  width: 100%;
+}
```
Comment (Leuchak) [inline src/components/JumbotronVideoComponent.vue:111]: split is already defined in split.scss

File: src/styles/split.scss [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+* Horizontal splitting
+*/
+.split-horizontal {
+flex-direction: column;
+}
+.split-horizontal > * {
+min-width: 100%;
+max-height: 50%;
+overflow-x: hidden;
+}
```
Comment (Leuchak) [inline src/styles/split.scss:34]: the original css was 
```css
.split-video {
  width: 50%;
}
.left-video {
  left: 0;
}
.right-video {
  right: 0;
}
```
I'd need a more detailed explanation of the reasoning behind these classes.
`.split` can also have a detrimental effect on content.

File: src/views/HomeView.vue [hunk 8]
```diff
@@ -14,7 +14,7 @@ import partenariatFile from '@clubcapra/assets/documents/CAPRA_PlanPartenariat.p
 </script>
 
 <template>
-  <JumbotronVideoComponent title="CAPRA" :video="introTrimmed" />
+  <JumbotronVideoComponent title="CAPRA" :videos="[introTrimmed]" />
```
Comment (Leuchak) [inline src/views/HomeView.vue:17]: unrelated change here.

File: src/components/JumbotronVideoComponent.vue [hunk 9]
```diff
@@ -24,18 +30,30 @@ if (props.img) {
...
+        loop
+      >
+        <source :src="video" type="video/mp4" />
+      </video>
+    </div>
   </div>
 </template>
 
-<style>
+<style lang="scss">
```
Comment (Leuchak) [inline src/components/JumbotronVideoComponent.vue:56]: The purpose of scss is to use variables to make it easier to modify values. I don't think scss is necessary here.

**Review by Leuchak — CHANGES_REQUESTED** [review]
The KISS principle (Keep It Simple, Stupid) is a fundamental concept in programming that I frequently apply when working on projects. Essentially, it reminds us to keep things simple and straightforward.

The idea behind KISS is that simplicity often leads to better understanding and easier code maintenance. When I begin to notice that my code is becoming complex, with too many layers, unnecessary functions, or convoluted structures, it's a warning sign for me.

That's when I put the KISS principle into action. I step back, simplify the code as much as possible, and look for ways to make it more elegant and easier to follow. The key is to keep things as simple as possible without sacrificing necessary functionality.

KISS is a foundational rule that helps us avoid over-engineering, reduce the risk of errors, and make our code more robust. It's an approach I favor to maintain code quality throughout a project's development.

File: src/components/JumbotronVideoComponent.vue (hunk 2 above)
Comment (patates-cipsi418) [reply src/components/JumbotronVideoComponent.vue:16]: I used this as reference https://tsdoc.org/pages/tags/defaultvalue/
but you mean this https://jsdoc.app/tags-param.html#optional-parameters-and-default-values.
Good
